cpp/CMakeLists.txt

```diff
@@ -215,18 +215,39 @@ else()
   set(PYTHON_EXECUTABLE ${Python3_EXECUTABLE})
 endif()
 
-if(ARROW_USE_CCACHE
-   AND NOT CMAKE_C_COMPILER_LAUNCHER
-   AND NOT CMAKE_CXX_COMPILER_LAUNCHER)
+if((ARROW_USE_SCCACHE)
+  AND NOT CMAKE_C_COMPILER_LAUNCHER
+  AND NOT CMAKE_CXX_COMPILER_LAUNCHER)
+
+  find_program(SCCACHE_FOUND sccache)
+  
+  if(NOT SCCACHE_FOUND AND DEFINED ENV{SCCACHE_PATH})
+    # cmake has problems finding sccache from within mingw
```

Comment:
Can we remove this by adding `chmod +x sccache/sccache.exe`?

Reply:
No it falls back to [ccache](https://github.com/ursacomputing/crossbow/runs/7388502838?check_suite_focus=true#step:7:5555) if we don't use the envvar hack. I would prefer to not need it too...